strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/HostedRepositoryProvider.java

```diff
@@ -71,4 +89,40 @@ public ArtifactOutputStream getOutputStream(String storageId,
         return (ArtifactOutputStream) Files.newOutputStream(repositoryPath);
     }
 
+    @Override
+    public List<Path> search(RepositorySearchRequest request)
+    {
+        List<Path> result = new LinkedList<Path>();
+        
+        Storage storage = configurationManager.getConfiguration().getStorage(request.getStorageId());
+        Repository repository = storage.getRepository(request.getRepositoryId());
+        LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
+        
+        List<ArtifactEntry> artifactEntryList = artifactEntryService.findByCoordinates(request.getStorageId(),
+                                                                                       request.getRepositoryId(),
+                                                                                       request.getCoordinates(),
+                                                                                       request.getSkip(),
+                                                                                       request.getLimit(),
+                                                                                       request.getOrderBy(), false);
+        
+        for (ArtifactEntry artifactEntry : artifactEntryList)
+        {
+            RepositoryPath repositoryPath;
+            try
+            {
+                repositoryPath = layoutProvider.resolve(repository, artifactEntry.getArtifactCoordinates());
+            }
+            catch (IOException e)
+            {
+                logger.error(String.format("Failed to resolve Remote Artifact [%s]", artifactEntry.getArtifactPath()),
```

Comment:
`Failed to resolve Remote Artifact` Why `Remote` ? It's hosted repository.

Reply:
right
thanks